test/new-e2e/tests/process/linux_test.go

```diff
@@ -0,0 +1,100 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package process
+
+import (
+	"testing"
+	"time"
+
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+)
+
+type linuxTestSuite struct {
+	e2e.Suite[e2e.FakeIntakeEnv]
+}
+
+func TestLinuxTestSuite(t *testing.T) {
+	e2e.Run(t, &linuxTestSuite{},
+		e2e.FakeIntakeStackDef(
+			e2e.WithAgentParams(agentparams.WithAgentConfig(processCheckConfigStr))))
+}
+
+func (s *linuxTestSuite) SetupSuite() {
+	s.Suite.SetupSuite()
+
+	// Start a process and keep it running
+	s.Env().VM.Execute("sudo apt-get -y install stress")
+	s.Env().VM.Execute("nohup stress -d 1 >myscript.log 2>&1 </dev/null &")
+}
+
+func (s *linuxTestSuite) TestProcessCheck() {
+	s.UpdateEnv(e2e.FakeIntakeStackDef(
+		e2e.WithAgentParams(agentparams.WithAgentConfig(processCheckConfigStr))))
+
+	t := s.T()
+
+	assert.EventuallyWithT(t, func(collect *assert.CollectT) {
+		assertRunningChecks(collect, s.Env().VM, []string{"process", "rtprocess"}, false)
+	}, 1*time.Minute, 5*time.Second)
+
+	var payloads []*aggregator.ProcessPayload
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		var err error
+		payloads, err = s.Env().Fakeintake.GetProcesses()
+		assert.NoError(c, err, "failed to get process payloads from fakeintake")
+		assert.NotEmpty(c, payloads, "no process payloads returned")
+	}, 2*time.Minute, 10*time.Second)
+
+	assertStressProcessCollected(t, payloads, false)
+}
+
+func (s *linuxTestSuite) TestProcessDiscoveryCheck() {
+	s.UpdateEnv(e2e.FakeIntakeStackDef(
+		e2e.WithAgentParams(agentparams.WithAgentConfig(processDiscoveryCheckConfigStr))))
+
+	t := s.T()
+
+	assert.EventuallyWithT(t, func(collect *assert.CollectT) {
+		assertRunningChecks(collect, s.Env().VM, []string{"process_discovery"}, false)
+	}, 1*time.Minute, 5*time.Second)
+
+	var payloads []*aggregator.ProcessDiscoveryPayload
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		var err error
+		payloads, err = s.Env().Fakeintake.GetProcessDiscoveries()
+		assert.NoError(c, err, "failed to get process discovery payloads from fakeintake")
+		assert.NotEmpty(c, payloads, "no process discovery payloads returned")
+	}, 2*time.Minute, 10*time.Second)
+
+	assertStressProcessDiscoveryCollected(t, payloads)
+}
+
+func (s *linuxTestSuite) TestProcessCheckWithIO() {
```

Comment:
Are these tests meant to eventually replace the manual QA steps? If so, could they also include checking logs for errors and running a manual check?

Reply:
Yes, totally!

1. I will add a test for manual checks in a separate PR
2. For logs, we currently do have error logs because the fake intake doesn't return a response (which we expect), so we'll have to update it with responses.

Reply:
Gotcha, thanks!